databuilder/databuilder/models/table_column_usage.py

```diff
@@ -6,64 +6,51 @@
 )
 
 from amundsen_rds.models import RDSModel
-from amundsen_rds.models.table import TableUsage as RDSTableUsage
 
 from databuilder.models.graph_node import GraphNode
 from databuilder.models.graph_relationship import GraphRelationship
 from databuilder.models.graph_serializable import GraphSerializable
 from databuilder.models.table_metadata import TableMetadata
 from databuilder.models.table_serializable import TableSerializable
-from databuilder.models.user import User
+from databuilder.models.usage.usage import Usage
 
 
-class ColumnReader(object):
+class ColumnReader(Usage):
     """
-    A class represent user's read action on column. Implicitly assumes that read count is one.
+    Represent user's read action on a table - and eventually on a column.
     """
 
     def __init__(self,
                  database: str,
                  cluster: str,
                  schema: str,
                  table: str,
-                 column: str,
+                 column: str,  # not used: per-column usage not yet implemented
                  user_email: str,
                  read_count: int = 1
                  ) -> None:
-        self.database = database
-        self.cluster = cluster
-        self.schema = schema
-        self.table = table
-        self.column = column
-        self.user_email = user_email
-        self.read_count = int(read_count)
 
-    def __repr__(self) -> str:
-        return f"ColumnReader(database={self.database!r}, cluster={self.cluster!r}, " \
-               f"schema={self.schema!r}, table={self.table!r}, column={self.column!r}, " \
-               f"user_email={self.user_email!r}, read_count={self.read_count!r})"
+        Usage.__init__(
+            self,
+            start_label=TableMetadata.TABLE_NODE_LABEL,
+            start_key=TableMetadata.TABLE_KEY_FORMAT.format(
+                db=database,
+                cluster=cluster,
+                schema=schema,
+                tbl=table),
+            user_email=user_email,
+            read_count=read_count,
+        )
 
 
 class TableColumnUsage(GraphSerializable, TableSerializable):
```

Comment:
Again, the naming is confusing and I am not sure why it's not just `TableReader`

Reply:
which one is for tableReader? i think the initial implementation is to implement both column usage and table usage in one model with query parser extractor with antlar. Then we found the query parser antlar has many production issues (e.g memory OOM) which we remove the extractor and leave the models here.